pandas/core/strings.py

```diff
@@ -449,11 +450,19 @@ def f(x):
             return [np.nan if item is None else item for item in m.groups()]
         else:
             return empty_row
+
     if regex.groups == 1:
-        result = Series([f(val)[0] for val in arr],
-                        name=_get_single_group_name(regex),
-                        index=arr.index, dtype=object)
+        if isinstance(arr, Index):
+            result = Index([f(val)[0] for val in arr],
+                           name=_get_single_group_name(regex),
+                           dtype=object)
+        else:
+            result = Series([f(val)[0] for val in arr],
+                            name=_get_single_group_name(regex),
+                            index=arr.index, dtype=object)
     else:
+        if isinstance(arr, Index):
```

Comment:
@jreback @shoyer Returning a `MultiIndex` is definitely a good idea, I think it was proposed here in a conversation thread https://github.com/pydata/pandas/pull/9870#issuecomment-92416184 but no issue has been created yet. I'll create one and link it back